language/stdlib/modules/vasp.move

```diff
@@ -133,225 +74,54 @@ module VASP {
-        Transaction::assert(is_root_vasp(root_account_addr), 7001);
-        // Child cert must be published under the child, but not yet certified
-        let child_vasp = AccountType::account_metadata<ChildVASP>(addr);
-        child_vasp.is_certified = true;
-        AccountType::update<ChildVASP>(addr, child_vasp);
-    }
-
     ///////////////////////////////////////////////////////////////////////////
     // Publicly callable APIs
     ///////////////////////////////////////////////////////////////////////////
 
-    // Publish the child VASP resource under the sending account. It can
-    // only be certified by the root account or a parent account under the
-    // same root VASP.
-    public fun apply_for_child_vasp_credential(root_vasp_addr: address) {
-        let sender = Transaction::sender();
-        Transaction::assert(!is_vasp(sender), 7002);
-        Transaction::assert(is_root_vasp(root_vasp_addr), 7001);
-        AccountType::assert_has_transition_cap<ChildVASP>(root_vasp_addr);
-        AccountType::apply_for<ChildVASP>(ChildVASP{ is_certified: false }, root_vasp_addr);
-    }
-
-    // The account at address `addr` is a root VASP if the account at that
-    // address has a certified AccountType<RootVASP> resource that has not expired.
-    public fun is_root_vasp(addr: address): bool {
-        if (AccountType::is_a<RootVASP>(addr)) {
-            let root_vasp = AccountType::account_metadata<RootVASP>(addr);
-            !root_credential_expired(&root_vasp)
-        } else {
-            false
-        }
-    }
-
-    // The account at address `addr` is a child VASP if it:
-    // 1. Is a (certified) ChildVASP account type; and
-    // 2. Its root_address field points to a valid (in the sense of
-    // `is_root_vasp`) root VASP account.
-    public fun is_child_vasp(addr: address): bool {
-        if (AccountType::is_a<ChildVASP>(addr)) {
-            is_root_vasp(AccountType::root_address<ChildVASP>(addr)) &&
-            AccountType::account_metadata<ChildVASP>(addr).is_certified
-        } else false
-    }
-
-    // Return the root address for a child VASP account.
-    public fun root_vasp_address(addr: address): address {
-        if (is_child_vasp(addr)) {
-            AccountType::root_address<ChildVASP>(addr)
-        } else if (is_root_vasp(addr)) {
-            addr
-        } else {
-            abort 2
-        }
-    }
-
-    // Return whether `root_vasp_addr` and `child_address` have a valid
-    // root-child VASP relationship.
-    public fun is_root_child_vasp(root_vasp_addr: address, child_address: address): bool {
-        if (is_root_vasp(root_vasp_addr) && is_child_vasp(child_address)) {
-            AccountType::root_address<ChildVASP>(child_address) == root_vasp_addr
-        } else {
-            false
-        }
+    // Create a child VASP resource for the sender
+    // Aborts if the sender is not a RootVASP
```

Comment:
So we no longer have the concept of a "parent account" for a VASP with this. Just want to confirm that that's OK.